### PR TITLE
Fixes for AB-SEC-001 to AB-SEC-011

### PR DESCRIPTION
**Pull request by ThomasBallandras**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,13 @@ configuration.
 
 You can install custom apache modules by adding a `.apache-mods` file at the root
 of your project. It must include one module per line.
+The list of modules available, with a short description of each module, is available at 
+the following locations depending on the version of the stack used:
+* Stack ubuntu-20: https://packages.ubuntu.com/search?suite=focal&searchon=names&keywords=libapache2-mod
```
Comment (Soulou) [inline README.md]:
```suggestion
* Stack `scalingo-20`: https://packages.ubuntu.com/search?suite=focal&searchon=names&keywords=libapache2-mod
```

File: README.md [hunk 2]
```diff
@@ -11,6 +11,13 @@ configuration.
...
+* Stack ubuntu-18: https://packages.ubuntu.com/search?suite=bionic&searchon=names&keywords=libapache2-mod
```
Comment (Soulou) [inline README.md]:
```suggestion
* Stack `scalingo-18`: https://packages.ubuntu.com/search?suite=bionic&searchon=names&keywords=libapache2-mod
```

File: bin/startup.sh [hunk 3]
```diff
@@ -12,43 +14,83 @@ if grep -q "libapache2-mod-auth-mellon" "${HOME}/Aptfile" ; then
...
+    exit 1
+  fi
+
   echo "Include ${HOME}/vendor/apache2/conf/site.conf" >> "${HOME}/vendor/apache2/conf/httpd.conf"
 fi
 
 # Starting
 echo "Starting Apache..."
-$HOME/.apt/usr/sbin/apache2 -f $HOME/vendor/apache2/conf/httpd.conf -DFOREGROUND
+"${HOME}"/.apt/usr/sbin/apache2 -f "${HOME}"/vendor/apache2/conf/httpd.conf -DFOREGROUND
```
Comment (Soulou) [inline bin/startup.sh]:
```suggestion
"${HOME}/.apt/usr/sbin/apache2" -f "${HOME}/vendor/apache2/conf/httpd.conf" -DFOREGROUND
```

File: bin/startup.sh [hunk 4]
```diff
@@ -1,8 +1,10 @@
 #!/bin/bash
 
 # Setting Mellon key, cert and metadata files
-if grep -q "libapache2-mod-auth-mellon" "${HOME}/Aptfile" ; then
+if grep -q "auth-mellon" "${HOME}/.apache-mods" ; then
   
+  export APACHE_WORKER_SIZE="${APACHE_WORKER_SIZE:="10"}"
```
Comment (Soulou) [inline bin/startup.sh]: Why only if mellon is present? It should be all the time no?

File: bin/startup.sh [hunk 5]
```diff
@@ -12,43 +14,83 @@ if grep -q "libapache2-mod-auth-mellon" "${HOME}/Aptfile" ; then
...
   fi
 fi
 
 # Compiling conf file
 echo "compiling conf file..."
 erb "${HOME}/vendor/apache2/conf/httpd.conf.erb" > "${HOME}/vendor/apache2/conf/httpd.conf"
 
+# Setting MaxRequestWorkers value according to the container size. Formula: total_mem / apache_process_mem_used (roughly 12mb)
+case "${CONTAINER_SIZE}" in
+  "S" ) SERVER_LIMIT="$(echo "256 / ${APACHE_WORKER_SIZE}" | "${HOME}/.apt/usr/bin/bc")"
```
Comment (Soulou) [inline bin/startup.sh]: Please don't use BC: example

```suggestion
  "S" ) SERVER_LIMIT="$(( 256 *APACHE_WORKER_SIZE} ))"
```

File: conf/httpd.conf.erb [hunk 6]
```diff
@@ -5,8 +5,7 @@ User appsdeck
 Group appsdeck
 PidFile <%= ENV["HOME"] %>/vendor/apache2/logs/apache2.pid
 
-Timeout 10
-MaxRequestWorkers 100
+Timeout 59
```
Comment (Soulou) [inline conf/httpd.conf.erb:9]: Please add a comment to explain why

**Review by Soulou — CHANGES_REQUESTED** [review]

File: bin/startup.sh (hunk 4 above)
Comment (ThomasBallandras) [reply bin/startup.sh]: Mistake - fixed

File: README.md (hunk 1 above)
Comment (ThomasBallandras) [reply README.md]: Done

File: README.md (hunk 2 above)
Comment (ThomasBallandras) [reply README.md]: Done

File: bin/startup.sh (hunk 3 above)
Comment (ThomasBallandras) [reply bin/startup.sh]: Done

File: bin/startup.sh (hunk 5 above)
Comment (ThomasBallandras) [reply bin/startup.sh]: Done - all fixed

File: conf/httpd.conf.erb (hunk 6 above)
Comment (ThomasBallandras) [reply conf/httpd.conf.erb:9]: Done

**Review by Soulou — APPROVED** [review]

**Pull request merged**